### PR TITLE
React Native 0.22 + Basic routing

### LINKED ISSUE #35: Routing
As a user sometimes I would like to see a map instead of AR. But sometimes I would like to see AR instead of a map.

### PR DESCRIPTION
**Pull request by bignell**
First, upgraded React Native to `0.22.2`. Post-`npm install`, make sure to run `react-native upgrade` and then `rnpm link`.

### Routing

A more step-by-step approach: first, basic routing through state based on @jake-shasteen's comment.

#### Notes:
- ListView apparently doesn't re-render unless `datasource` changes
    - The best solution at present seems to be cloning `datasource` in `componentWillMount`
- Removed SideMenu from the app root with the goal of moving to ViewContainer in the future

Resolves #35

### CODE REVIEW

**Comment by jake-shasteen** [comment]
Pulled and tested this PR on a different machine and phone to make sure it runs and doesn't seem to introduce other bugs.

File: app/components/ViewContainer.js [hunk 1]
```diff
@@ -48,22 +49,41 @@ export default class ViewContainer extends Component {
...
+  toggleView(view) {
+    this.setState({ view });
+  }
+
   render() {
     const { getLocationToSave, pins, deletePin, recent } = this.props;
     return (
       <View style={{flex: 1}}>
-        <AR currLoc={ this.state.currLoc } pins={pins} />
+      { this.state.view === 'ar' ? <AR currLoc={ this.state.currLoc } pins={pins} /> : void 0 }
```
Comment (tiffeli) [inline app/components/ViewContainer.js]: thanks for refactoring! This makes things much more D.R.Y. :+1:

File: index.ios.js [hunk 2]
```diff
@@ -29,27 +27,10 @@ const store = createStore(
...
-        <SideMenu menu={menu}>
-          <ViewContainer >
-          </ViewContainer>
-        </SideMenu>
-      </Provider>
-    );
-  }
-}
+const findAR = () => (
+  <Provider store={store}>
```
Comment (tiffeli) [inline index.ios.js]: we took out side menu?

**Comment by tiffeli** [comment]
@bignell @jake-shasteen Great job guys. TODO:  We will still need that router from the former pull request, but to route between view and non view pages (settings..etc).  But not all those features are done yet so it's a todo for now. I think this looks good. Works on my phone when I tested.

**Comment by asu91** [comment]
@bignell @jake-shasteen Bravo :heavy_check_mark:

**Pull request merged**